### PR TITLE
Widget Test: MinyChip

### PR DESCRIPTION
**Pull request by krushnakantanayak**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-72

### Description

- Selected MinyChip
-  Unselected MinyChip
- Toggle Selection
- OnSelected Callback

### Screenshots

miny_chip_test

### Tips

Any tips for testing it?

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [ ] All items in the author checklist is checked or not applicable
- [ ] All files including .lock and assets are reviewed
- [ ] Reasonable UI flows/cases related to the change are manually tested
- [ ] No commented-out code or unnecessary debug statements.
- [ ] No spelling or grammatical errors in code comments or documentation.
- [ ] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [ ] The branch is up-to-date with the target branch (e.g., staging).
- [ ] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

File: test/miny_chip_test.dart [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                onSelected: onSelected,
+              ),
+            ),
+          ),
+        );
+
+    testWidgets(
+      'Given MinyChip is selected '
+      'When the widget is rendered '
+      'Then it should have primary color as background, light color for text',
```
Comment (quiteachaos) [inline test/miny_chip_test.dart]: it should have contrastDark as bg color not primary

File: test/miny_chip_test.dart [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        final textWidget = tester.widget<Text>(find.text(label));
+        expect(textWidget.style!.color, appTheme.colors.light);
+      },
+    );
+
+    testWidgets(
+      'Given MinyChip is not selected '
+      'When the widget is rendered '
+      'Then its secondary color as background, contrastDark color for text',
```
Comment (quiteachaos) [inline test/miny_chip_test.dart]: contrastLow as bg color

File: test/miny_chip_test.dart [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+        final textWidget = tester.widget<Text>(find.text(label));
+        expect(textWidget.style!.color, appTheme.colors.contrastDark);
+      },
+    );
+
+    testWidgets(
+      'Given MinyChip with onSelected callback '
+      'When it is tapped '
+      'Then it should toggle the background color between primary & secondary',
```
Comment (quiteachaos) [inline test/miny_chip_test.dart]: between contrastDark and contrastLow

File: test/miny_chip_test.dart [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+          onSelected: (_) {},
+        ));
+
+        // Assert
+        final container = tester.widget<Container>(find.byType(Container));
+        expect(
+          (container.decoration as BoxDecoration).color,
+          appTheme.colors.contrastDark,
+        );
+
```
Comment (quiteachaos) [inline test/miny_chip_test.dart:49]: also expect that tick icon is visible when selected

File: test/miny_chip_test.dart [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+          onSelected: (_) {},
+        ));
+
+        // Assert
+        final container = tester.widget<Container>(find.byType(Container));
+        expect(
+          (container.decoration as BoxDecoration).color,
+          appTheme.colors.contrastLow,
+        );
+
```
Comment (quiteachaos) [inline test/miny_chip_test.dart:77]: also expect that tick icon is not visible

File: test/miny_chip_test.dart [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        await tester.tap(find.text(label));
+        await tester.pumpAndSettle();
+
+        // Assert:
+        var container = tester.widget<Container>(find.byType(Container));
+        expect(
+          (container.decoration as BoxDecoration).color,
+          appTheme.colors.contrastDark,
+        );
+
```
Comment (quiteachaos) [inline test/miny_chip_test.dart:110]: check if tick icon is also visible or not

**Review by quiteachaos — CHANGES_REQUESTED** [review]

File: test/miny_chip_test.dart (hunk 1 above)
Comment (krushnakantanayak) [reply test/miny_chip_test.dart]: done

File: test/miny_chip_test.dart (hunk 2 above)
Comment (krushnakantanayak) [reply test/miny_chip_test.dart]: done

File: test/miny_chip_test.dart (hunk 3 above)
Comment (krushnakantanayak) [reply test/miny_chip_test.dart]: done

File: test/miny_chip_test.dart (hunk 4 above)
Comment (krushnakantanayak) [reply test/miny_chip_test.dart:49]: done

File: test/miny_chip_test.dart (hunk 5 above)
Comment (krushnakantanayak) [reply test/miny_chip_test.dart:77]: done

File: test/miny_chip_test.dart (hunk 6 above)
Comment (krushnakantanayak) [reply test/miny_chip_test.dart:110]: done

**Review by quiteachaos — APPROVED** [review]

**Pull request merged**